src/settings/Validation/loan-history/general.test.js

```diff
@@ -0,0 +1,91 @@
+import general from './general';
+
+import { closedLoansRules } from '../../../constants';
+
+describe('general', () => {
+  describe('data methods', () => {
+    const isClosingTypeIntervalSelected = jest.fn(() => true);
+    const isAnyClosingTypeSelected = jest.fn(() => true);
+    const testData = {
+      isClosingTypeIntervalSelected,
+      isAnyClosingTypeSelected,
+    };
+
+    beforeEach(() => {
+      general(testData);
+    });
+
+    afterEach(() => {
+      isClosingTypeIntervalSelected.mockClear();
+      isAnyClosingTypeSelected.mockClear();
+    });
+
+    Object.values(closedLoansRules).forEach(rule => {
+      describe(`for ${rule}`, () => {
+        it('should call isClosingTypeIntervalSelected with correct props', () => {
+          expect(isClosingTypeIntervalSelected).toHaveBeenCalledWith(rule);
+        });
+
+        it('should call isClosingTypeIntervalSelected twice', () => {
+          const callNumber = isClosingTypeIntervalSelected.mock.calls
+            .filter(item => item[0] === rule)
+            .length;
+
+          expect(callNumber).toBe(2);
+        });
+
+        it('should call isAnyClosingTypeSelected with correct props', () => {
+          expect(isAnyClosingTypeSelected).toHaveBeenCalledWith(rule);
+        });
+      });
+    });
+  });
+
+  describe('when "isClosingTypeIntervalSelected" in passed "data" returns true and "isAnyClosingTypeSelected" return false', () => {
+    const testData = {
+      isClosingTypeIntervalSelected: () => true,
+      isAnyClosingTypeSelected: () => false,
```

Comment:
On 7-th and 8-th lines you use jest.fn. 
Could you choose one approach?

Reply:
On the 7-th and 8-th line I need to have access for `mock.calls`.
I don't need it here.
Should I still add `jest.fn` ?

Reply:
ok, let it be